src/node_sockaddr-inl.h

```diff
@@ -166,6 +166,23 @@ bool SocketAddress::operator!=(const SocketAddress& other) const {
   return !(*this == other);
 }
 
+bool SocketAddress::operator<(const SocketAddress& other) const {
+  return this->compare(other) == -1;
+}
+
+bool SocketAddress::operator>(const SocketAddress& other) const {
+  return this->compare(other) == 1;
+}
+
+bool SocketAddress::operator<=(const SocketAddress& other) const {
+  int c = this->compare(other);
+  return c == -2 ? false : c <= 0;
```

Comment:
`compare` should probably return an `enum` or at least named values, similar to [`std::partial_ordering`](https://en.cppreference.com/w/cpp/utility/compare/partial_ordering), because I think generally the expectation would be that compare behaves like a standard comparison function (e.g. `memcmp()`) based on its signature

Also, the `this->` prefixes can all be dropped.

Reply:
Yeah, I was considering the `enum` option also. Happy to get the confirmation ;-) ... I'll switch it to that.